cephfs/file_test.go

```diff
@@ -615,3 +615,95 @@ func TestFallocate(t *testing.T) {
 		assert.Error(t, err)
 	})
 }
+
+func TestFlock(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("validate", func(t *testing.T) {
+		f := &File{}
+		err := f.Flock(LockSH, 1010)
+		assert.Error(t, err)
+	})
+
+	t.Run("validateOperation", func(t *testing.T) {
+		fname := "file.txt"
+		f, err := mount.Open(fname, os.O_RDWR|os.O_CREATE, 0666)
+		assert.NoError(t, err)
+		assert.NotNil(t, f)
+		defer func() {
+			assert.NoError(t, f.Close())
+			assert.NoError(t, mount.Unlink(fname))
+		}()
+		err = f.Flock(1037, 1010)
+		assert.Error(t, err)
+	})
+
+	t.Run("basicLocking", func(t *testing.T) {
+
+		fname1 := "file1.txt"
+		f1, err := mount.Open(fname1, os.O_RDWR|os.O_CREATE, 0666)
+		assert.NoError(t, err)
+		assert.NotNil(t, f1)
+		defer func() {
+			assert.NoError(t, f1.Close())
+			assert.NoError(t, mount.Unlink(fname1))
+		}()
+		// Lock exclusively twice.
+		t.Run("exclusiveTwiceBlock", func(t *testing.T) {
+			err := f1.Flock(LockEX, 42)
```

Comment:
as you might need different owners, use constants for the `42` and `43` values please. That makes it a little clearer how the tests work. Call one `anna` and the other `bob` or something :smiley: